mlflow/utils/autologging_utils/safety.py

```diff
@@ -573,7 +558,54 @@ def call_original(*og_args, **og_kwargs):
                 if original_has_been_called:
                     return original_result
                 else:
-                    return original(*args, **kwargs)
+                    try:
+                        try_log_autologging_event(
+                            AutologgingEventLogger.get_logger().log_original_function_start,
+                            session,
+                            destination,
+                            function_name,
+                            args,
+                            kwargs,
+                        )
+                        original_result = original(*args, **kwargs)
+                        try_log_autologging_event(
+                            AutologgingEventLogger.get_logger().log_original_function_success,
+                            session,
+                            destination,
+                            function_name,
+                            args,
+                            kwargs,
+                        )
+                        return original_result
+                    except Exception as e:
+                        try_log_autologging_event(
+                            AutologgingEventLogger.get_logger().log_original_function_error,
+                            session,
+                            destination,
+                            function_name,
+                            args,
+                            kwargs,
+                            e,
+                        )
+                        failed_during_original = True
+                        raise
+                    finally:
+                        if not failed_during_original:
+                            try_log_autologging_event(
+                                AutologgingEventLogger.get_logger().log_patch_function_error,
+                                session,
+                                destination,
+                                function_name,
+                                args,
+                                kwargs,
+                                patch_function_exception,
+                            )
```

Comment:
Shouldn't we only do this if `patch_function_exception` is not `None`?  Can we add test coverage to make sure we aren't emitting this event unless there's a patch function exception?